source/characterProcessing.py

```diff
@@ -519,6 +519,33 @@ def __init__(self, locale):
 			log.error("Invalid complex symbol regular expression in locale %s: %s" % (locale, e))
 			raise LookupError
 
+	def _replaceGroups(self, m, string):

```

Comment:
Rather than in the doc string, please add this as a type annotation.
```suggestion
	def _replaceGroups(self, m: re.Matchobject, string: str) -> str:

```

Reply:
Ok, done so